Makefile

```diff
@@ -54,9 +54,10 @@ venv: requirements.txt ## Prepare the virtual environment
 	test -d $(VENV) || virtualenv -q --python=$(PYTHON) $(VENV);\
 	source $(VENV)/bin/activate || exit 1;\
 	pip install -q -r requirements.txt;\
-	touch $(VENV);\
+	touch $(VENV);
 
 lint: venv  ## Lint the project
+	source $(VENV)/bin/activate; \
```

Comment:
L55 does the activate -> https://github.com/elastic/apm-integration-testing/pull/1086/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R55

Do you think the `venv` goal could be split in two, then the consumers don't need to do anything else?


```
prepare-venv: requirements.txt ## Prepare the virtual environment
	test -d $(VENV) || virtualenv -q --python=$(PYTHON) $(VENV);

venv: prepare-venv  ## Prepare the virtual environment
	source $(VENV)/bin/activate || exit 1;\
	pip install -q -r requirements.txt;\
	touch $(VENV);
```

Reply:
I do not see advantages

Reply:
You don't need to change the consumers, so the other goals remain as they are

Reply:
I just tested it, have the same issue, it does not activate the environment correctly in some cases. I have tried these two codes:

```
prepare-venv: requirements.txt ## Prepare the virtual environment
	test -d $(VENV) || virtualenv -q --python=$(PYTHON) $(VENV);

venv: prepare-venv  ## Prepare the virtual environment
	source $(VENV)/bin/activate || exit 1;\
	pip install -q -r requirements.txt;\
	touch $(VENV);
```

```
prepare-venv: requirements.txt ## Prepare the virtual environment
	test -d $(VENV) || virtualenv -q --python=$(PYTHON) $(VENV);

venv: prepare-venv  ## Prepare the virtual environment
	source $(VENV)/bin/activate || exit 1;\
	pip install -q -r requirements.txt;\
	touch $(VENV);\
```

I think is better to be explicit and avoid magic, when magic stop working you do not realize what's going on.